gmso/external/convert_dataframe.py

```diff
@@ -0,0 +1,552 @@
+            dataList[i] = [float(x) for x in dataList[i]]  # turn to float
+    return dataList
+
+
+def _parse_unyts_to_headers(dataList, columns) -> list:
+    new_colsList = []
+    for data, col in zip(dataList, columns):
+        if isinstance(data, u.unyt_array):
+            unit = str(
+                data[0].units
+            )  # assumption that all data in List is same units
+            new_colsList.append(col + f"({unit})")
+        else:
+            new_colsList.append(col)
+    return new_colsList
+
+
+def _generate_component_lists(topology, parameter, columns) -> list:
+    outList = []
+    columnsList = []
+    for column in columns:
+        valuesList = _recursive_getattr(topology, parameter, column)
+        if isinstance(valuesList[0], dict):
+            # add keys to columnsList and values to outList
+            keys = list(valuesList[0].keys())
+            values_dictList = [
+                [value[key] for value in valuesList] for key in keys
+            ]
+            outList.extend(values_dictList)
+            columnsList.extend(keys)
+        elif isinstance(valuesList[0], u.unyt_array) and not isinstance(
+            valuesList[0], u.unyt_quantity
+        ):
+            outList.extend(np.array(valuesList).T)
+            if column == "position":
+                columnsList.extend(["x", "y", "z"])
+            else:
+                columnsList.extend(
+                    [f"{column}-({i})" for i in range(len(valuesList[0]))]
+                )
+        elif isinstance(valuesList[0], tuple) or isinstance(
+            valuesList[0], list
+        ):  # could be connection_members
+            outList.extend(np.array(valuesList).T)
+            if "connection_members" in column:
+                columnsList.extend(
+                    [
+                        f"{parameter} member ({i})"
+                        for i in range(len(valuesList[0]))
+                    ]
+                )
+            else:
+                columnsList.extend(
+                    [f"{column}-({i})" for i in range(len(valuesList[0]))]
+                )
+
+            # handle positions?
+            # handle connection_members
+            pass
+        else:
+            outList.append(valuesList)
+            columnsList.append(column)
+    return outList, columnsList
+
+
+def _recursive_getattr(topology, attr, attr_attr):
+    """Parse a topology to get a list of attributes from an iterable."""
+
+    def _getattr(obj, attr1):
+        try:
+            return getattr(obj, attr1)
+        except AttributeError as e:
+            raise AttributeError(
+                f"The GMSO Topology is missing the requested attribute {attr1} from {obj}.{attr_attr}"
+            )
+
+    iteritems = getattr(topology, attr)
+    parseFunction = lambda x: functools.reduce(
+        _getattr, [x] + attr_attr.split(".")
+    )
+    return list(map(parseFunction, iteritems))
+
+
+def _pandas_from_parameters(
```

Comment:
Is this function called or used anywhere? Looks like this is the old method from `Topology`, I'm guessing we can remove it here?